lib/bolt/executor.rb

```diff
@@ -241,52 +245,73 @@ def with_node_logging(description, batch)
 
     def run_command(targets, command, options = {})
       description = options.fetch(:description, "command '#{command}'")
+      # TODO: Consider creating a run_command_* plan function that accepts a block
+      # that returns a target-specific command.
+      target_mapping = targets.product([command]).to_h
```

Comment:
This use of `product` is a little mind-boggling but I think it's fine. Another option would be `targets.zip([command]*targets.length)` but I don't know that it's any clearer.

Reply:
Fwiw, I also looked at `each_with_object` and `zip`. Using `product` seemed like the most concise way to accomplish what will hopefully be a temporary workaround.